src/server/server.rs

```diff
@@ -451,13 +452,25 @@ impl<T: RaftStoreRouter, S: StoreAddrResolver> Server<T, S> {
     }
 
     fn on_snap_token(&mut self, event_loop: &mut EventLoop<Self>, _: u64, token: Token) {
-        // Now we close every snapshot connection after sending successfully.
+        // We use new connection for every snapshot, and will close it after sending OK.
+        // But we can't close it here directly, because we can't guarantee that
```

Comment:
Actually not, tcp doesn't guarantee that sending buffered data before when we call close, even using `flush` only flushes the data to system socket buffer. 

Maybe we should use shutdown first, and then close it some time later.